src/main.ts

```diff
@@ -29,19 +29,25 @@ export function activate(context: vscode.ExtensionContext): any {
 
     let logger = new Logger(text => _channel.append(text));
 
-    ensureRuntimeDependencies(extension, logger, reporter)
-        .then((success : boolean) => {
-            // activate language services
-            OmniSharp.activate(context, reporter, _channel);
-
-            // register JSON completion & hover providers for project.json
-            context.subscriptions.push(addJSONProviders());
-            
-            if (success) {
-                // activate coreclr-debug
-                coreclrdebug.activate(extension, context, reporter, logger, _channel);
-            }
-        });
+    let runtimeDependenciesExist = await ensureRuntimeDependencies(extension, logger, reporter);
+
+    if (!runtimeDependenciesExist) {
+        //do something
+    }
+    
+    // activate language services
+    let omniSharpPromise = OmniSharp.activate(context, reporter, _channel);
+
+    // register JSON completion & hover providers for project.json
+    context.subscriptions.push(addJSONProviders());
+    
+    // activate coreclr-debug
+    let coreClrDebugPromise = coreclrdebug.activate(extension, context, reporter, logger, _channel);
+
+    return {
+        initializationFinished: Promise.all([omniSharpPromise, coreClrDebugPromise])
+        .then(a => {})
```

Comment:
Is the `.then(a => {})` part required? If so, out of curiosity -- why?

Reply:
the effect of `.then(a => {})` is to hide the value of `a`. 

I wanted to expose a Thenable which resolves once our extension is fully initialized [in addition to activated]. I did not intend to leak the internal implementation of that Thenable, so I'm using this statement to hide Promise.all's default result value.

Reply:
That sounds very commentable :smile:

Reply:
Ah, okay.

Reply:
Added comment.